packages/js/src/plugins/auctionHouseModule/createBid.ts

```diff
@@ -163,19 +167,39 @@ export const createBidBuilder = async (
   };
 
   // Sell Instruction.
-  // ToDo: Add support for the auctioneerAuthority
   let buyInstruction;
-  if (tokenAccount) {
-    buyInstruction = createBuyInstruction({ ...accounts, tokenAccount }, args);
-  } else {
-    buyInstruction = createPublicBuyInstruction(
-      { ...accounts, tokenAccount: buyerTokenAccount },
-      args
+  if (params.auctioneerAuthority) {
```

Comment:
We should probably add another check here to make sure that `auctionHouse.hasAuctioneer` is `true`.

If `auctionHouse.hasAuctioneer` is `true` AND `params.auctioneerAuthority` is not provided, we should throw a custom error explaining what went wrong. See examples in `errors.ts`.

Could you do the same for `createListing` please?